### PR TITLE
feat(httpendpoint): add tests for cli

### PR DESCRIPTION
**Pull request by sicoyle**
# Description

Add checks for new Dapr resource httpendpoints for external service invocation as per comment [here](https://github.com/dapr/dapr/pull/6227#pullrequestreview-1384871630).

## Issue reference
https://github.com/dapr/dapr/pull/6227

Please reference the issue this PR will close: [#_[4549]_](https://github.com/dapr/dapr/issues/4549)

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [X] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Comment by sicoyle** [comment]
Hey @mukundansundar!
Do you have any pointers on e2e tests for v1.11.0 features? I added the external service invocation functionality with the new HTTPEndpoint Dapr resource, and am having a hard time with configuring the e2e tests for it.

Does the runtime v1.11.0 release have to be cut before those would pass? I feel like this is almost a chicken before the egg scenario here...  I saw the CRDs work off the [raw data for the specific runtime version yaml files](https://github.com/dapr/cli/blob/76bf893295dc6d025fb4eacbbbf2abc831141185/pkg/kubernetes/upgrade.go#L161) so I tried a work around pointing to master for the HTTPEndpoint yaml file, but then it tried to look for a "master" helm-chart release which does not exist. I also see TODO comments in `common.go` stating `TODO: Skip the dashboard version check for now until the helm chart is updated`, so that had me thinking that I also need the updated helm chart.

**Comment by pravinpushkar** [comment]
This will depend on latest runtime changes and will need runtime RC at-least.

**Comment by sicoyle** [comment]
> This will depend on latest runtime changes and will need runtime RC at-least.

Thank you @pravinpushkar!! I will circle back here then when the release is ready!

**Comment by pravinpushkar** [comment]
@sicoyle dapr RC is out. Probably this PR can make use of that.

**Comment by mukundansundar** [comment]
> @sicoyle dapr RC is out. Probably this PR can make use of that.

I think 1284 needs to be merged before that.

**Comment by sicoyle** [comment]
E2E self hosted failure is due to timing out on installing podman:
```
Waiting for VM ...
Error: The action has timed out.
```

also upload test results failure shows:
```
Run actions/upload-artifact@master
Error: Input required and not supplied: path
```

**Comment by sicoyle** [comment]
I think there may be a github issue or workflow issue going on for that last build because looking through the E2E upgrade path tests I see it's failing because of timing issues again:

Going to `common.go:792` is from a 2 minute timeout that expired waiting for pods to delete.
```
goroutine 786 [chan send, 18 minutes]:
github.com/dapr/cli/tests/e2e/common.waitPodDeletion(0xc0004ccfd0?, 0xf51f2a?, 0x0?)
	/home/runner/work/cli/cli/src/github.com/dapr/cli/tests/e2e/common/common.go:1031 +0x1f1
created by github.com/dapr/cli/tests/e2e/common.uninstallTest.func1
	/home/runner/work/cli/cli/src/github.com/dapr/cli/tests/e2e/common/common.go:792 +0x198
```

**Comment by sicoyle** [comment]
documenting again bc I've seen this issue inconsistently as well in my many e2e test failures 😅  The inconsistency part is definitely a timing related issue we should capture and address in the future; however, for the time being, this should have a conditional since the dapr dashboard is removed from the helm chart in v1.11 RC [here](https://github.com/dapr/dapr/pull/6151). I am opening a PR to address separately.

```
=== RUN   TestUpgradePathNonHAModeMTLSDisabled/vv1.11.0-rc.4_to_v1.10.5/clusterroles_exist_v1.11.0-rc.4
    common.go:454: 
        	Error Trace:	/home/runner/work/cli/cli/src/github.com/dapr/cli/tests/e2e/common/common.go:454
        	Error:      	Not equal: 
        	            	expected: true
        	            	actual  : false
        	Test:       	TestUpgradePathNonHAModeMTLSDisabled/vv1.11.0-rc.4_to_v1.10.5/clusterroles_exist_v1.11.0-rc.4
        	Messages:   	cluster role dapr-dashboard, found = false, wanted = true
```

File: tests/e2e/upgrade/upgrade_test.go [hunk 1]
```diff
@@ -94,6 +94,22 @@ var supportedUpgradePaths = []upgradePath{
 			CustomResourceDefs:  []string{"components.dapr.io", "configurations.dapr.io", "subscriptions.dapr.io", "resiliencies.dapr.io"},
 		},
 	},
+	{
```
Comment (yaron2) [inline tests/e2e/upgrade/upgrade_test.go:97]: This test case should be removed (`"v1.11.0-rc.4"` to `"1.10.5"`)

File: tests/e2e/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -289,8 +297,60 @@ func TestUpgradePathHAModeMTLSEnabled(t *testing.T) {
...
+	for _, p := range supportedUpgradePaths {
+		t.Run(fmt.Sprintf("setup v%s to v%s", p.previous.RuntimeVersion, p.next.RuntimeVersion), func(t *testing.T) {
+			t.Run("delete CRDs "+p.previous.RuntimeVersion, common.DeleteCRD(p.previous.CustomResourceDefs))
+			t.Run("delete CRDs "+p.next.RuntimeVersion, common.DeleteCRD(p.next.CustomResourceDefs))
+		})
+	}
+
+	for _, p := range supportedUpgradePaths {
+		// only check runtime versions that support HTTPEndpoint resource.
+		if p.next.RuntimeVersion != "v1.11.0-rc.4" {
```
Comment (yaron2) [inline tests/e2e/upgrade/upgrade_test.go:332]: Please change to `if strings.Contains(p.next.RuntimeVersion, "1.11")` to support every 1.11.x version

**Review by yaron2 — CHANGES_REQUESTED** [review]

**Pull request merged**